Remove leading Countrycode from EU-VAT-Numbers

Drischie opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios,
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description

EU-VAT-Numbers has always leading Countrycodes. To validate VAT-Number we have to remove it before sending the request.
After we've checked if country is in eu we will remove the Countrycode and send the request.

### Fixed Issues (if relevant)

EU-VAT-Numbers always invalid cause of leading Countrycodes.

### Manual testing scenarios

EU-VAT-Numbers looks like DEXXXXXXXXX for a german one.
Before change:
Giving DE 293 778 674 in Configuration->General->Store Information or in creating an Account will resullt in "Invalid VAT".
After change:
Giving DE 293 778 674 in Configuration->General->Store Information or in creating an Account will resullt in "VAT-Number is valid".

### Contribution checklist
 - [x ] Pull request has a meaningful description of its purpose
 - [x ] All commits are accompanied by meaningful commit messages
 - [x ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x ] All automated tests passed successfully (all builds on Travis CI are green)

Drischie committed:
Remove leading Countrycode from EU-VAT-Numbers

EU-VAT-Numbers has always leading Countrycodes. To validate VAT-Number we have to remove it before sending the request.
After we've checked if country is in eu we will remove the Countrycode and send the request.

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=17385) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=17385) before we can accept your contribution.<br/>

magento-engcom-team commented:
Hi @Drischie. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

magento-engcom-team commented:
@Drischie thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

orlangur commented on app/code/Magento/Customer/Model/Vat.php:
```diff
@@ -184,9 +184,9 @@ public function checkVatNumber($countryCode, $vatNumber, $requesterCountryCode =
             $requestParams = [];
             $requestParams['countryCode'] = $countryCode;
-            $requestParams['vatNumber'] = str_replace([' ', '-'], ['', ''], $vatNumber);
+            $this->isCountryInEU($countryCode) ? $requestParams['vatNumber'] = str_replace([' ', '-', $countryCode], ['', '', ''], $vatNumber) : $requestParams['vatNumber'] = str_replace([' ', '-'], ['', ''], $vatNumber);
```
Always put variable like `$requestParams['vatNumber']` out of ternary operator.

Drischie committed:
Removed $requestParams[] from ternary Operator

Moved variable $requestParams[] out of ternary operator.
To do this, the new variables $vatNumberSanitized and $requesterVatNumberSanitized are introduced.

Drischie committed:
Changed ternary to multiline

Changed one-line ternary to multiline ternary to not exceed line-size-limit of 120 Characters.

hostep commented:
Small remark here, the VAT number sanitisation is only done for validating against the [VIES service](http://ec.europa.eu/taxation_customs/vies/), right?

So this means that when customers enter their VAT number with a country prefix or without a country prefix, both will be seen as valid (very good, I'm all for it!). But both will be stored in the database in a different format.

I'm just not sure if shop owners would appreciate inconsistently formatted VAT numbers?
This is just a remark, hoping to spark some discussion.

Drischie committed:
Shorten Variable Name requesterVatNumberSanitized

Changed newly introduced variable VatNumberSanitized to reqVatNumSanitized to meet coding standards.

Drischie commented:
@hostep You're right but VAT-Numbers with always be different in the Database cause they are different =D
EU-Vat have always leading Countrycodes, others don't. So i have to save EU-Vat in Database including Countrycode to show me the right one but i have to remove Countrycode to check against VIES.

hostep commented:
Indeed, but if VAT numbers can be entered with or without the country code, shouldn't it make sense that we try to save them consistently in the database as well. So they either **all** have the country code or **none** have the country code. And maybe also with spaces and dashes removed, so they all are formatted in the same way in the database?

I'm not saying it *should* happen, this is only a remark to see what other people think about this.

(Additionally, I just remembered this [very old ticket](https://github.com/magento/magento2/issues/1251) I once created a long time ago, and the solution in this PR kind of solves point number 3 mentioned in there 😉)

Commit:
Minor fixes for better code readability

Commit:
Code style fixes

magento-engcom-team commented:
Hi @Drischie. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.7. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

roman204 commented:
We did also develop a complete module to fix this behavior, including handling of countries like Greek and many more.
https://github.com/CopeX/CopeX_VatFix
